src/test/java/com/github/dariobalinzo/task/ElasticSourceTaskTest.java

```diff
@@ -150,9 +150,61 @@ public void shouldRunSourceTaskWithJsonCastFilter() throws IOException, Interrup
         //when (fetching first page)
         task.start(conf);
         List<SourceRecord> poll1 = task.poll();
-        assertEquals("Struct{fullName=\"Test\",age=10,ts=111}", poll1.get(0).value().toString());
+        assertEquals("Struct{fullName=\"Test\",nonavrofield=non-avro-field,avroField=avro-field,age=10,ts=111}", poll1.get(0).value().toString());
 
         task.stop();
     }
 
+    @Test
+    public void shouldRunSourceTaskWithAvroNameConverter() throws IOException, InterruptedException {
+        //given
+        deleteTestIndex();
+
+        insertMockData(111);
+        insertMockData(112);
+        insertMockData(113);
+        insertMockData(114);
+        refreshIndex();
+
+        ElasticSourceTask task = new ElasticSourceTask(ElasticSourceConnectorConfig.AVRO_FIELD_CONVERTER);
+        Mockito.when(context.offsetStorageReader()).thenReturn(MockOffsetFactory.empty());
+        task.initialize(context);
+        Map<String, String> conf = getConf();
+        conf.put(ElasticSourceConnectorConfig.FIELDS_JSON_CAST_CONFIG, "fullName");
```

Comment:
please avoid to enable FIELDS_JSON_CAST_CONFIG that is not relevant for this test. The getConf() method will return the "standard" conf for the test, you just need to put in the conf your new config for avro convertion.